### ISSUE TITLE
No temperature values if chip frequency is under 160 MHz

### ISSUE THREAD

**Issue by pixelfriese** [pstolarz/Arduino-Temperature-Control-Library#15]
Hi,

I use an ESP32 (Wrover) and the current library together with OneWireNg in version 0.11.2. 

When I lower the frequency of the chip below 160 MHz, I don´t get any values from the DS18B20 temperature sensor. If the frequency is at 160 MHz or above, I get valid values. This is the case in the example scripts and in my own scripts. 

The following line is used to set the frequency in the setup function of the arduino code:

Works:
`setCpuFrequencyMhz(160);`

Failed to read:
`setCpuFrequencyMhz(80);`

**Comment by pstolarz** [comment]
For performance reason OneWireNg library uses `F_CPU` macro-define to detect configured CPU freq instead of call to `getCpuFrequencyMhz()`. So if you want to use the lib on a specific frequency set this define correctly while building the lib. If you want to change frequency dynamically during the program run, don't use `CONFIG_BITBANG_DELAY_CCOUNT` mode (comment it out in `OneWireNg_Config.h`).

**Comment by pstolarz** [comment]
And please don't use this repo for reporting issues for OneWireNg. Use [the proper place](https://github.com/pstolarz/OneWireNg).

**Comment by pstolarz** [comment]
I've updated OneWIreNg lib to get rid of `F_CPU` usage. Now the frequency is detected on runtime. For performance reason the detection is done only once, that is the frequency is read at initial time and the read value is used subsequently. For this reason, if you change the frequency  by `setCpuFrequencyMhz()`, do this before first read-out of 1-wire sensor by the library. This way the library will detect the frequency correctly in the initial frequency read. If you'd like to change the frequency on runtime, call `ccntUpdateCpuFreqMHz()` routine to inform OneWIreNg about the update.

I've checked the new version of library (currently on master branch) on all my ESP32 boards + ESP8266 and everything works fine. If you had any issues with this change, report the issue for OneWireNg.
